// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/floating-window/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FloatingWindow = void 0;
var enums_1 = require("./enums");
// FloatingWindow Extensible Area
var FloatingWindow = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the Floating Window
     *
     * @param top - position in which the the top left corner of the floating window is relative
     * to the top of the rendered window. It must return the root element where the floating window
     * was rendered.
     * @param left - position in which the the top left corner of the floating window is relative
     * to the left of the rendered window.
     * @param movable - tells whether the floating window is movable or static.
     * @param backgroundColor - background color of the floating window.
     * @param boxShadow - box shadow to apply to the floating window
     * @param contentFunction - function that gives the html element to render the content of
     * the floating window. It must return the root element where the floating window was rendered.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function FloatingWindow(_a) {
        var id = _a.id, top = _a.top, left = _a.left, movable = _a.movable, backgroundColor = _a.backgroundColor, boxShadow = _a.boxShadow, contentFunction = _a.contentFunction;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "FloatingWindow_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.top = top;
        this.left = left;
        this.movable = movable;
        this.backgroundColor = backgroundColor;
        this.boxShadow = boxShadow;
        this.contentFunction = contentFunction;
        this.type = enums_1.FloatingWindowType.CONTAINER;
    }
    return FloatingWindow;
}());
exports.FloatingWindow = FloatingWindow;
