// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/user-list-item-additional-information/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListItemLabel = exports.UserListItemIcon = void 0;
var enums_1 = require("./enums");
// UserListItemAdditionalInformation Extensible Area
var UserListItemIcon = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the User List Item Additional information Item.
     * In this case, a icon.
     *
     * @param icon - icon to be used in the user list item additional information.
     * It goes on the left side of it.
     * @param userId - the userId in which this information will appear when the user
     * list item is clicked.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function UserListItemIcon(_a) {
        var id = _a.id, _b = _a.icon, icon = _b === void 0 ? '' : _b, _c = _a.userId, userId = _c === void 0 ? '' : _c;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserListItemIcon_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.icon = icon;
        this.userId = userId;
        this.type = enums_1.UserListItemAdditionalInformationType.ICON;
    }
    return UserListItemIcon;
}());
exports.UserListItemIcon = UserListItemIcon;
var UserListItemLabel = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the User List Item Additional information Item.
     * In this case, a label (Information).
     *
     * @param label - text to be displayed in the user list item.
     * @param icon - icon to be used in the user list item additional information.
     * It goes on the left side of it.
     * @param userId - the userId in which this information will appear when the user
     * list item is clicked.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function UserListItemLabel(_a) {
        var id = _a.id, _b = _a.icon, icon = _b === void 0 ? '' : _b, _c = _a.userId, userId = _c === void 0 ? '' : _c, _d = _a.label, label = _d === void 0 ? '' : _d;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserListItemLabel_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.icon = icon;
        this.label = label;
        this.userId = userId;
        this.type = enums_1.UserListItemAdditionalInformationType.LABEL;
    }
    return UserListItemLabel;
}());
exports.UserListItemLabel = UserListItemLabel;
