// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/options-dropdown-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptionsDropdownSeparator = exports.OptionsDropdownOption = void 0;
var enums_1 = require("./enums");
// OptionsDropdown Extensible Area
var OptionsDropdownOption = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the Options Dropdown. In this case,
     * an option (Similar to a button, the user can click it).
     *
     * @param label - label to be displayed in the options dropdown option.
     * @param icon - icon to be displayed in the options dropdown. It goes in the left side of it.
     * @param onClick - function to be called when clicking the option.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function OptionsDropdownOption(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.onClick, onClick = _d === void 0 ? function () { } : _d;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "OptionsDropdownOption_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.label = label;
        this.icon = icon;
        this.onClick = onClick;
        this.type = enums_1.OptionsDropdownItemType.OPTION;
    }
    return OptionsDropdownOption;
}());
exports.OptionsDropdownOption = OptionsDropdownOption;
var OptionsDropdownSeparator = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the Navigation Bar. In this case,
     * a separator.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function OptionsDropdownSeparator() {
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "OptionsDropdownSeparator_".concat(id);
        };
        this.type = enums_1.OptionsDropdownItemType.SEPARATOR;
    }
    return OptionsDropdownSeparator;
}());
exports.OptionsDropdownSeparator = OptionsDropdownSeparator;
