// bigbluebutton-html-plugin-sdk/dist/cjs/ui-commands/commands.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.uiCommands = void 0;
var commands_1 = require("./chat/commands");
var commands_2 = require("./external-video/commands");
var commands_3 = require("./sidekick-options-container/commands");
var commands_4 = require("./presentation-area/commands");
var commands_5 = require("./user-status/commands");
var commands_6 = require("./conference/commands");
var commands_7 = require("./notification/commands");
var commands_8 = require("./camera/commands");
var commands_9 = require("./actions-bar/commands");
var commands_10 = require("./layout/commands");
var commands_11 = require("./nav-bar/commands");
var commands_12 = require("./sidekick-area/commands");
exports.uiCommands = {
    actionsBar: commands_9.actionsBar,
    camera: commands_8.camera,
    chat: commands_1.chat,
    externalVideo: commands_2.externalVideo,
    sidekickOptionsContainer: commands_3.sidekickOptionsContainer,
    sidekickArea: commands_12.sidekickArea,
    navBar: commands_11.navBar,
    presentationArea: commands_4.presentationArea,
    userStatus: commands_5.userStatus,
    conference: commands_6.conference,
    notification: commands_7.notification,
    layout: commands_10.layout,
};
