// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/camera-settings-dropdown-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CameraSettingsDropdownSeparator = exports.CameraSettingsDropdownOption = void 0;
var enums_1 = require("./enums");
// CameraSettingsDropdown Extensible Area
var CameraSettingsDropdownOption = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the camera settings dropdown. In this case,
     * an option.
     *
     * @param label - label to be displayed in camera settings dropdown option.
     * @param icon - icon to be used in the option for the dropdown. It goes in the left side of it.
     * @param dataTest - data-test attribute to be used in the option for the dropdown.
     * @param onClick - function to be called when clicking the button.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function CameraSettingsDropdownOption(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.dataTest, dataTest = _d === void 0 ? '' : _d, _e = _a.onClick, onClick = _e === void 0 ? function () { } : _e;
        var _this = this;
        this.id = '';
        this.dataTest = '';
        this.setItemId = function (id) {
            _this.id = "CameraSettingsDropdownOption_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.label = label;
        this.icon = icon;
        this.dataTest = dataTest;
        this.onClick = onClick;
        this.type = enums_1.CameraSettingsDropdownItemType.OPTION;
    }
    return CameraSettingsDropdownOption;
}());
exports.CameraSettingsDropdownOption = CameraSettingsDropdownOption;
var CameraSettingsDropdownSeparator = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the camera settings dropdown. In this case,
     * a separator.
     *
     * @remarks
     * It will display a horizontal thin black line inside the dropdown.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5)
     */
    function CameraSettingsDropdownSeparator() {
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "CameraSettingsDropdownSeparator_".concat(id);
        };
        this.type = enums_1.CameraSettingsDropdownItemType.SEPARATOR;
    }
    return CameraSettingsDropdownSeparator;
}());
exports.CameraSettingsDropdownSeparator = CameraSettingsDropdownSeparator;
