// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/user-list-dropdown-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListDropdownTitleAction = exports.UserListDropdownGenericContentInformation = exports.UserListDropdownFixedContentInformation = exports.UserListDropdownSeparator = exports.UserListDropdownOption = void 0;
var enums_1 = require("./enums");
// UserListDropdown Extensible Area
var UserListDropdownOption = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the User List Dropdown. In this case,
     * a button.
     *
     * @param label - label to be displayed in user list dropdown option.
     * @param tooltip - label to be displayed when hovering the user list dropdown option.
     * @param icon - icon to be used in the user list dropdown option. It goes in the left side of it.
     * @param onClick - function to be called when clicking the option.
     * @param allowed - if false, the use list dropdown  will not appear in the dropdown.
     * @param userId - the userId in which this dropdown option will appear when the user
     * list item is clicked.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function UserListDropdownOption(_a) {
        var _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.tooltip, tooltip = _d === void 0 ? '' : _d, _e = _a.allowed, allowed = _e === void 0 ? true : _e, _f = _a.onClick, onClick = _f === void 0 ? function () { } : _f, _g = _a.userId, userId = _g === void 0 ? '' : _g;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserListDropdownOption_".concat(id);
        };
        this.userId = userId;
        this.label = label;
        this.icon = icon;
        this.tooltip = tooltip;
        this.allowed = allowed;
        this.onClick = onClick;
        this.type = enums_1.UserListDropdownItemType.OPTION;
    }
    return UserListDropdownOption;
}());
exports.UserListDropdownOption = UserListDropdownOption;
var UserListDropdownSeparator = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the User List Dropdown. In this case,
     * a separator.
     *
     * @param userId - the userId in which this dropdown separator will appear when the user
     * list item is clicked.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function UserListDropdownSeparator(_a) {
        var _b = _a.userId, userId = _b === void 0 ? '' : _b, _c = _a.position, position = _c === void 0 ? enums_1.UserListDropdownSeparatorPosition.AFTER : _c;
        var _this = this;
        this.id = '';
        this.position = enums_1.UserListDropdownSeparatorPosition.AFTER;
        this.setItemId = function (id) {
            _this.id = "UserListDropdownSeparator_".concat(id);
        };
        this.userId = userId;
        this.position = position;
        this.type = enums_1.UserListDropdownItemType.SEPARATOR;
    }
    return UserListDropdownSeparator;
}());
exports.UserListDropdownSeparator = UserListDropdownSeparator;
var UserListDropdownFixedContentInformation = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the User List Dropdown. In this case,
     * a button.
     *
     * @param label - label to be displayed in user list dropdown information.
     * @param icon - icon to be used in the user list dropdown information.
     * It goes on the left side of it.
     * @param iconRight - icon to be used in the user list dropdown information.
     * It goes on the right side of it.
     * @param allowed - if false, the use list dropdown  will not appear in the dropdown.
     * @param textColor - Color that the text will have.
     * @param userId - the userId in which this dropdown information will appear when the user
     * list item is clicked.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function UserListDropdownFixedContentInformation(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.iconRight, iconRight = _d === void 0 ? '' : _d, _e = _a.allowed, allowed = _e === void 0 ? true : _e, _f = _a.userId, userId = _f === void 0 ? '' : _f, _g = _a.textColor, textColor = _g === void 0 ? '' : _g;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserListDropdownFixedContentInformation_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.userId = userId;
        this.label = label;
        this.icon = icon;
        this.iconRight = iconRight;
        this.textColor = textColor;
        this.allowed = allowed;
        this.type = enums_1.UserListDropdownItemType.FIXED_CONTENT_INFORMATION;
    }
    return UserListDropdownFixedContentInformation;
}());
exports.UserListDropdownFixedContentInformation = UserListDropdownFixedContentInformation;
var UserListDropdownGenericContentInformation = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the User List Dropdown. In this case,
     * a button.
     *
     * @param label - label to be displayed in user list dropdown information.
     * @param icon - icon to be used in the user list dropdown information.
     * It goes on the left side of it.
     * @param iconRight - icon to be used in the user list dropdown information.
     * It goes on the right side of it.
     * @param allowed - if false, the use list dropdown  will not appear in the dropdown.
     * @param textColor - Color that the text will have.
     * @param userId - the userId in which this dropdown information will appear when the user
     * list item is clicked.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function UserListDropdownGenericContentInformation(_a) {
        var id = _a.id, contentFunction = _a.contentFunction, _b = _a.allowed, allowed = _b === void 0 ? true : _b, _c = _a.userId, userId = _c === void 0 ? '' : _c;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserListDropdownGenericContentInformation_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.userId = userId;
        this.allowed = allowed;
        this.contentFunction = contentFunction;
        this.type = enums_1.UserListDropdownItemType.GENERIC_CONTENT_INFORMATION;
    }
    return UserListDropdownGenericContentInformation;
}());
exports.UserListDropdownGenericContentInformation = UserListDropdownGenericContentInformation;
var UserListDropdownTitleAction = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the User List Dropdown. In this case,
     * a button (action) for the title.
     *
     * @param onClick - onClick function.
     * @param tooltip - quick explanatory message that will appear when user hover the action.
     * @param icon - icon to be used in the user list dropdown title action.
     * It goes on the left side of it.
     * @param userId - the userId in which this dropdown title action will appear when the user
     * list item is clicked.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function UserListDropdownTitleAction(_a) {
        var id = _a.id, _b = _a.icon, icon = _b === void 0 ? '' : _b, _c = _a.userId, userId = _c === void 0 ? '' : _c, onClick = _a.onClick, tooltip = _a.tooltip;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserListDropdownTitleAction_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.userId = userId;
        this.icon = icon;
        this.tooltip = tooltip;
        this.onClick = onClick;
        this.type = enums_1.UserListDropdownItemType.TITLE_ACTION;
    }
    return UserListDropdownTitleAction;
}());
exports.UserListDropdownTitleAction = UserListDropdownTitleAction;
