// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/user-list-dropdown-item/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserListDropdownSeparatorPosition = exports.UserListDropdownItemType = void 0;
// User list dropdown items types:
var UserListDropdownItemType;
(function (UserListDropdownItemType) {
    UserListDropdownItemType["OPTION"] = "USER_LIST_DROPDOWN_OPTION";
    UserListDropdownItemType["SEPARATOR"] = "USER_LIST_DROPDOWN_SEPARATOR";
    UserListDropdownItemType["FIXED_CONTENT_INFORMATION"] = "USER_LIST_DROPDOWN_FIXED_CONTENT_INFORMATION";
    UserListDropdownItemType["GENERIC_CONTENT_INFORMATION"] = "USER_LIST_DROPDOWN_GENERIC_CONTENT_INFORMATION";
    UserListDropdownItemType["TITLE_ACTION"] = "USER_LIST_DROPDOWN_TITLE_ACTION";
})(UserListDropdownItemType || (exports.UserListDropdownItemType = UserListDropdownItemType = {}));
var UserListDropdownSeparatorPosition;
(function (UserListDropdownSeparatorPosition) {
    UserListDropdownSeparatorPosition["BEFORE"] = "BEFORE_DEFAULT_ITEMS";
    UserListDropdownSeparatorPosition["AFTER"] = "AFTER_DEFAULT_ITEMS";
})(UserListDropdownSeparatorPosition || (exports.UserListDropdownSeparatorPosition = UserListDropdownSeparatorPosition = {}));
