// bigbluebutton-html-plugin-sdk/dist/cjs/data-channel/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataChannelTypes = exports.DataChannelHooks = exports.DataChannelPushEntryFunctionUserRole = void 0;
// Role
/**
 * Enum for each role in possible for a user to assume in the meeting
 * The developer can specify them when pushing data with the push function
 * returned from the `pluginApi.useChannel`.
 */
var DataChannelPushEntryFunctionUserRole;
(function (DataChannelPushEntryFunctionUserRole) {
    DataChannelPushEntryFunctionUserRole["PRESENTER"] = "PRESENTER";
    DataChannelPushEntryFunctionUserRole["MODERATOR"] = "MODERATOR";
    DataChannelPushEntryFunctionUserRole["VIEWER"] = "VIEWER";
})(DataChannelPushEntryFunctionUserRole || (exports.DataChannelPushEntryFunctionUserRole = DataChannelPushEntryFunctionUserRole = {}));
var DataChannelHooks;
(function (DataChannelHooks) {
    DataChannelHooks["DATA_CHANNEL_BUILDER"] = "Hooks::DataChannelBuilder";
    DataChannelHooks["DATA_CHANNEL_RESET"] = "Hooks::DataChannelReset";
    DataChannelHooks["DATA_CHANNEL_DELETE"] = "Hooks::DataChannelDelete";
    DataChannelHooks["DATA_CHANNEL_REPLACE"] = "Hooks::DataChannelReplace";
})(DataChannelHooks || (exports.DataChannelHooks = DataChannelHooks = {}));
var DataChannelTypes;
(function (DataChannelTypes) {
    DataChannelTypes["ALL_ITEMS"] = "Hooks::DataChannel::AllItems";
    DataChannelTypes["NEW_ITEMS"] = "Hooks::DataChannel::NewItems";
    DataChannelTypes["LATEST_ITEM"] = "Hooks::DataChannel::LatestItem";
})(DataChannelTypes || (exports.DataChannelTypes = DataChannelTypes = {}));
