// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/presentation-toolbar-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PresentationToolbarSeparator = exports.PresentationToolbarSpinner = exports.PresentationToolbarButton = void 0;
var enums_1 = require("./enums");
// PresentationToolbar Extensible Area
var PresentationToolbarButton = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for presentation toolbar. In this case
     * a button.
     *
     * @param label - label to be displayed in the button
     * @param tooltip - tooltip to be displayed when hovering the button
     * @param onClick - function to be called when clicking the button
     * @param style - style of the button in the presentation toolbar
     * @param dataTest - data-test attribute for testing purposes
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5)
     */
    function PresentationToolbarButton(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.tooltip, tooltip = _c === void 0 ? '' : _c, _d = _a.dataTest, dataTest = _d === void 0 ? '' : _d, _e = _a.onClick, onClick = _e === void 0 ? function () { } : _e, _f = _a.style, style = _f === void 0 ? {} : _f;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "PresentationToolbarButton_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.label = label;
        this.tooltip = tooltip;
        this.onClick = onClick;
        this.style = style;
        this.dataTest = dataTest;
        this.type = enums_1.PresentationToolbarItemType.BUTTON;
    }
    return PresentationToolbarButton;
}());
exports.PresentationToolbarButton = PresentationToolbarButton;
var PresentationToolbarSpinner = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for presentation toolbar. In this case
     * a spinner (ring that will be rotating).
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5)
     */
    function PresentationToolbarSpinner() {
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "PresentationToolbarSpinner_".concat(id);
        };
        this.type = enums_1.PresentationToolbarItemType.SPINNER;
    }
    return PresentationToolbarSpinner;
}());
exports.PresentationToolbarSpinner = PresentationToolbarSpinner;
var PresentationToolbarSeparator = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for presentation toolbar. In this case
     * a separator.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5)
     */
    function PresentationToolbarSeparator() {
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "PresentationToolbarSeparator_".concat(id);
        };
        this.type = enums_1.PresentationToolbarItemType.SEPARATOR;
    }
    return PresentationToolbarSeparator;
}());
exports.PresentationToolbarSeparator = PresentationToolbarSeparator;
