// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/action-button-dropdown-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionButtonDropdownSeparator = exports.ActionButtonDropdownOption = void 0;
var enums_1 = require("./enums");
// ActionButtonDropdown Extensible Area
var ActionButtonDropdownOption = /** @class */ (function () {
    /**
     * Returns the option for the action button dropdown
     *
     * @param label - label to be displayed on the option
     * @param icon - icon to be displayed on the option
     * @param tooltip - tooltip to be displayed when hovering over option
     * @param dataTest - string attribute to be used for testing
     * @param allowed - boolean indicating whether the option should be displayed
     * @param onClick - function to be called when clicking
     *
     * @returns the option to be displayed in the action button dropdown
     */
    function ActionButtonDropdownOption(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.tooltip, tooltip = _d === void 0 ? '' : _d, _e = _a.dataTest, dataTest = _e === void 0 ? '' : _e, _f = _a.allowed, allowed = _f === void 0 ? true : _f, _g = _a.onClick, onClick = _g === void 0 ? function () { } : _g;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "ActionButtonDropdownOption_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.label = label;
        this.icon = icon;
        this.tooltip = tooltip;
        this.dataTest = dataTest;
        this.allowed = allowed;
        this.onClick = onClick;
        this.type = enums_1.ActionButtonDropdownItemType.OPTION;
    }
    return ActionButtonDropdownOption;
}());
exports.ActionButtonDropdownOption = ActionButtonDropdownOption;
var ActionButtonDropdownSeparator = /** @class */ (function () {
    /**
     * Returns the separator for the action button dropdown
     *
     * @param dataTest - optional string attribute to be used for testing
     *
     * @returns the separator to be displayed in the action button dropdown
     */
    function ActionButtonDropdownSeparator(_a) {
        var _b = _a === void 0 ? {} : _a, _c = _b.dataTest, dataTest = _c === void 0 ? '' : _c;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "ActionButtonDropdownSeparator_".concat(id);
        };
        this.type = enums_1.ActionButtonDropdownItemType.SEPARATOR;
        this.dataTest = dataTest;
    }
    return ActionButtonDropdownSeparator;
}());
exports.ActionButtonDropdownSeparator = ActionButtonDropdownSeparator;
