// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/nav-bar-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NavBarInfo = exports.NavBarButton = void 0;
var enums_1 = require("./enums");
// NavBar Extensible Area
var NavBarButton = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the Navigation Bar. In this case,
     * a button.
     *
     * @param label - label to be displayed in navigation bar button.
     * @param tooltip - label to be displayed when hovering the navigation bar button.
     * @param icon - icon to be used in the navigation bar button. It goes in the left side of it.
     * @param onClick - function to be called when clicking the button.
     * @param position - position to place the navigation bar button.
     * See {@link NavBarItemPosition}
     * @param hasSeparator - boolean indicating whether the navigation bar button has separator
     * (vertical bar)
     * @param disabled - if true, the navigation bar button will not be clickable
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function NavBarButton(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.tooltip, tooltip = _d === void 0 ? '' : _d, _e = _a.disabled, disabled = _e === void 0 ? true : _e, _f = _a.onClick, onClick = _f === void 0 ? function () { } : _f, _g = _a.position, position = _g === void 0 ? enums_1.NavBarItemPosition.RIGHT : _g, _h = _a.hasSeparator, hasSeparator = _h === void 0 ? true : _h;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "NavBarButton_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.label = label;
        this.icon = icon;
        this.tooltip = tooltip;
        this.disabled = disabled;
        this.onClick = onClick;
        this.type = enums_1.NavBarItemType.BUTTON;
        this.hasSeparator = hasSeparator;
        this.position = position;
    }
    return NavBarButton;
}());
exports.NavBarButton = NavBarButton;
var NavBarInfo = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the Navigation Bar. In this case,
     * an informative label.
     *
     * @param label - label to be displayed in navigation bar information.
     * @param position - position to place the navigation bar information.
     * See {@link NavBarItemPosition}
     * @param hasSeparator - boolean indicating whether the navigation bar information has separator
     * (vertical bar)
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function NavBarInfo(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.position, position = _c === void 0 ? enums_1.NavBarItemPosition.RIGHT : _c, _d = _a.hasSeparator, hasSeparator = _d === void 0 ? true : _d;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "NavBarInfo_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.label = label;
        this.type = enums_1.NavBarItemType.INFO;
        this.position = position;
        this.hasSeparator = hasSeparator;
    }
    return NavBarInfo;
}());
exports.NavBarInfo = NavBarInfo;
