// bigbluebutton-html-plugin-sdk/dist/cjs/ui-commands/layout/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeEnforcedLayoutTypeEnum = exports.EnforcedLayoutTypeEnum = exports.LayoutEnum = void 0;
var LayoutEnum;
(function (LayoutEnum) {
    LayoutEnum["CHANGE_ENFORCED_LAYOUT"] = "CHANGE_ENFORCED_LAYOUT";
    LayoutEnum["SET_ENFORCED_LAYOUT"] = "SET_ENFORCED_LAYOUT";
})(LayoutEnum || (exports.LayoutEnum = LayoutEnum = {}));
var EnforcedLayoutTypeEnum;
(function (EnforcedLayoutTypeEnum) {
    EnforcedLayoutTypeEnum["CUSTOM_LAYOUT"] = "CUSTOM_LAYOUT";
    EnforcedLayoutTypeEnum["SMART_LAYOUT"] = "SMART_LAYOUT";
    EnforcedLayoutTypeEnum["PRESENTATION_FOCUS"] = "PRESENTATION_FOCUS";
    EnforcedLayoutTypeEnum["VIDEO_FOCUS"] = "VIDEO_FOCUS";
    EnforcedLayoutTypeEnum["CAMERAS_ONLY"] = "CAMERAS_ONLY";
    EnforcedLayoutTypeEnum["PRESENTATION_ONLY"] = "PRESENTATION_ONLY";
    EnforcedLayoutTypeEnum["PARTICIPANTS_AND_CHAT_ONLY"] = "PARTICIPANTS_AND_CHAT_ONLY";
    EnforcedLayoutTypeEnum["MEDIA_ONLY"] = "MEDIA_ONLY";
})(EnforcedLayoutTypeEnum || (exports.EnforcedLayoutTypeEnum = EnforcedLayoutTypeEnum = {}));
exports.ChangeEnforcedLayoutTypeEnum = EnforcedLayoutTypeEnum;
