// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/presentation-dropdown-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PresentationDropdownSeparator = exports.PresentationDropdownOption = void 0;
var enums_1 = require("./enums");
// PresentationDropdown Extensible Area
var PresentationDropdownOption = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the Presentation Dropdown. In this case,
     * an option (clickable).
     *
     * @param label - label to be displayed in the presentation dropdown option.
     * @param icon - icon to be displayed in the presentation dropdown.
     * It goes in the left side of it.
     * @param onClick - function to be called when clicking the option.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function PresentationDropdownOption(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.onClick, onClick = _d === void 0 ? function () { } : _d;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "PresentationDropdownOption_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.label = label;
        this.icon = icon;
        this.onClick = onClick;
        this.type = enums_1.PresentationDropdownItemType.OPTION;
    }
    return PresentationDropdownOption;
}());
exports.PresentationDropdownOption = PresentationDropdownOption;
var PresentationDropdownSeparator = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for the Presentation Dropdown. In this case,
     * a separator (horizontal thin black line).
  
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5).
     */
    function PresentationDropdownSeparator() {
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "PresentationDropdownSeparator_".concat(id);
        };
        this.type = enums_1.PresentationDropdownItemType.SEPARATOR;
    }
    return PresentationDropdownSeparator;
}());
exports.PresentationDropdownSeparator = PresentationDropdownSeparator;
