// bigbluebutton-html-plugin-sdk/dist/cjs/core/enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GENERIC_HOOK_PLUGIN = exports.HookEvents = void 0;
var HookEvents;
(function (HookEvents) {
    HookEvents["BBB_CORE_SENT_NEW_DATA"] = "BBB_CORE_SENT_NEW_DATA";
    HookEvents["BBB_CORE_UPDATED_STATE"] = "BBB_CORE_UPDATED_STATE";
    HookEvents["BBB_CORE_UPDATED_MEETING_STATUS"] = "BBB_CORE_UPDATED_MEETING_STATUS";
    /**
     * PLUGIN_SENT_CHANGES_TO_BBB_CORE is used to:
     *  - Communicate that the subscription has changed for the dom-element-manipulation hook;
     *  - Send replace/delete event for useDataChannel;
     *  - Trigger the mutation function already created
     */
    HookEvents["PLUGIN_SENT_CHANGES_TO_BBB_CORE"] = "PLUGIN_SENT_CHANGES_TO_BBB_CORE";
    HookEvents["PLUGIN_SUBSCRIBED_TO_BBB_CORE"] = "PLUGIN_SUBSCRIBED_TO_BBB_CORE";
    HookEvents["PLUGIN_UNSUBSCRIBED_FROM_BBB_CORE"] = "PLUGIN_UNSUBSCRIBED_FROM_BBB_CORE";
})(HookEvents || (exports.HookEvents = HookEvents = {}));
exports.GENERIC_HOOK_PLUGIN = 'GENERIC_HOOK_PLUGIN';
