// bigbluebutton-html-plugin-sdk/dist/cjs/extensible-areas/user-camera-dropdown-item/component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserCameraDropdownSeparator = exports.UserCameraDropdownOption = void 0;
var enums_1 = require("./enums");
// UserCameraDropdown Extensible Area
var UserCameraDropdownOption = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for User Camera Dropdown. In this case
     * an option.
     *
     * @param label - label to be displayed in the option.
     * @param icon - icon to be displayed in the option. Left side of it.
     * @param onClick - function to be called when clicking the button
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5)
     */
    function UserCameraDropdownOption(_a) {
        var id = _a.id, _b = _a.label, label = _b === void 0 ? '' : _b, _c = _a.icon, icon = _c === void 0 ? '' : _c, _d = _a.onClick, onClick = _d === void 0 ? function () { } : _d, _e = _a.displayFunction, displayFunction = _e === void 0 ? function () { return true; } : _e;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserCameraDropdownOption_".concat(id);
        };
        if (id) {
            this.id = id;
        }
        this.displayFunction = displayFunction;
        this.label = label;
        this.icon = icon;
        this.onClick = onClick;
        this.type = enums_1.UserCameraDropdownItemType.OPTION;
    }
    return UserCameraDropdownOption;
}());
exports.UserCameraDropdownOption = UserCameraDropdownOption;
var UserCameraDropdownSeparator = /** @class */ (function () {
    /**
     * Returns object to be used in the setter for User Camera Dropdown. In this case
     * a separator.
     *
     * @returns Object that will be interpreted by the core of Bigbluebutton (HTML5)
     */
    function UserCameraDropdownSeparator(_a) {
        var _b = _a === void 0 ? { displayFunction: function () { return true; } } : _a, displayFunction = _b.displayFunction;
        var _this = this;
        this.id = '';
        this.setItemId = function (id) {
            _this.id = "UserCameraDropdownSeparator_".concat(id);
        };
        this.displayFunction = displayFunction;
        this.type = enums_1.UserCameraDropdownItemType.SEPARATOR;
    }
    return UserCameraDropdownSeparator;
}());
exports.UserCameraDropdownSeparator = UserCameraDropdownSeparator;
